crates/datafusion/src/lib.rs

```diff
@@ -287,4 +330,73 @@ mod tests {
             verify_data_with_replacecommits(&ctx, &sql, test_table.as_ref()).await
         }
     }
+
+    #[tokio::test]
+    async fn datafusion_read_external_hudi_table() {
+        for test_table in &[
+            V6ComplexkeygenHivestyle,
+            V6Nonpartitioned,
+            V6SimplekeygenNonhivestyle,
+            V6SimplekeygenHivestyleNoMetafields,
+            V6TimebasedkeygenNonhivestyle,
+        ] {
+            println!(">>> testing for {}", test_table.as_ref());
+            let ctx = prepare_session_context_with_table_factory().await;
+            let base_path = test_table.path();
+
+            let create_table_sql = format!(
+                "CREATE EXTERNAL TABLE {} STORED AS HUDITABLE LOCATION '{}'",
```

Comment:
i think `HUDITABLE` is a bit redundant - we already know it's creating a table. More importantly, in other sql engine like spark/flink/hive, we all use `HUDI` to indicate the format. So let's be consistent and simplify this as `HUDI`